libs/blocks/card/card.js

```diff
@@ -17,24 +18,38 @@ const getCardType = (styles) => {
   return cardTypes[authoredType] || HALF;
 };
 
+const getUpFromSectionMetadata = (section) => {
+  const sectionMetadata = getSectionMetadata(section.querySelector('.section-metadata'));
+  const styles = sectionMetadata.style?.split(', ').map((style) => style.replaceAll(' ', '-'));
+  return styles?.find((style) => style.includes('-up'));
+};
+
 const addWrapper = (el, section, cardType) => {
-  const { classList } = section;
   const gridCl = 'consonant-CardsGrid';
+  const prevGrid = section.querySelector(`.consonant-Wrapper .${gridCl}`);
 
-  if (!classList.contains('consonant-Wrapper')) {
-    const up = Array.from(classList).find((style) => style.includes('-up'))?.replace('-', '') || '3up';
-    const innerWrapper = `<div class="consonant-Wrapper-inner">
-      <div class="consonant-Wrapper-collection">
-        <div class="${gridCl} ${gridCl}--${up} ${gridCl}--with4xGutter${cardType === DOUBLE_WIDE ? ` ${gridCl}--doubleWideCards` : ''}">
-        </div>
-      </div>
-    </div>`;
-
-    classList.add('consonant-Wrapper', 'consonant-Wrapper--1200MaxWidth');
-    section.insertAdjacentHTML('afterbegin', innerWrapper);
+  if (prevGrid) {
+    prevGrid.append(el);
+    return;
   }
 
-  section.querySelector(`.${gridCl}`).append(el);
+  const upClass = getUpFromSectionMetadata(section);
+  const up = upClass?.replace('-', '') || '3up';
+  const gridClass = `${gridCl} ${gridCl}--${up} ${gridCl}--with4xGutter${cardType === DOUBLE_WIDE ? ` ${gridCl}--doubleWideCards` : ''}`;
+  const grid = createTag('div', { class: gridClass }, el);
+  const collection = createTag('div', { class: 'consonant-Wrapper-collection' }, grid);
+  const inner = createTag('div', { class: 'consonant-Wrapper-inner' }, collection);
+  const wrapper = createTag('div', { class: 'milo-card-wrapper consonant-Wrapper consonant-Wrapper--1200MaxWidth' }, inner);
+  const prevSib = el.previousElementSibling;
+  const nextSib = el.nextElementSibling;
+
+  if (prevSib) {
+    prevSib.after(wrapper);
```

Comment:
Consider adding /* c8 ignore next [3]*/ for tests.

Reply:
Good catch. I added a test for it instead